### DISCUSSION TITLE
[Feature Request] Administration of standing orders

### DISCUSSION THREAD

**Discussion by Teronox** [Zrips/CMI#276] [Feature Requests]
Dear Zrips,

i think your plugin is very good, but one feature would make it more awesome.

It would be very helpful and cool, if player could administrate standing orders.
For example, you wouldn't have to pay manual every week with "/pay playername amount" your rent or something, you would be able to make a standing order, e.g. "/standingorder playername interval amount".
So, if you rent claim blocks from another player, you could create a standing order to this player.

Command considerations:

/standingorder playername interval amount usage = pays every [interval] the [amount] to player [playername], the usage is the usage of the standing order.
(e.g.: /standingorder Teronox 1w 2500 Survivalclaim-rent)
(w = week, i guess useful would be the options to paying per day, week, month)
/standingorder list = shows all the standing orders with the [usage], [playername], amount and so one.
/standingorder del usage = delete the standingorder with the [usage] you will not continue (you don't want to pay anymore automaticly to the [playername] the [amount] of money).

What sees the standing order "aim"-player, the player which get the money:
He could get an ingame email, which shows [player which made the standing order] payed [amount] with usage: [usage], [interval]

What happens, if the player, which create the standing order, hasn't enough money anymore?
The destination-player of this standing order gets an email, which show's something like "player
[playername] hasn't enough money for the following standing order: [usage], [amount], [interval], the standing order was stopped."

I hope, that i could express my feature idea adequate.
(My english isn't very well)

Thanks in advance and best regards,
Teronox

**Comment by ScuroK** [comment]
thats a nice idea! it would allow players to do cool things like monthly payments. and if there is no money on the account the monthly payment would be stopped and a notification email would be sendt to the reciever.

nice concept!

**Comment by CloudeLecaw** [comment]
I like your idea too, Handy up for it :D

Maybe to keep it more simply just use an amount of days instead of 1w 1d 12hours   etc  or all in Hours..or so

**Comment by Teronox** [comment]
@CloudeLecaw Yeah i already mentioned the amount only in hours for the interval, but then you must convert by your own the aimduration from X years or X Months or X days to the amount of hours you must type in. ;)

**Comment by mrfloris** [comment]
Would it be possible now in 2020 to achieve this using specialized commands and scheduler.yml or eventcommands.yml?
